### PR TITLE
Build with emscripten 3.1.7

### PR DESCRIPTION
**Pull request by radekdoulik**
Add emscripten-version.txt file to the nuget package

### CODE REVIEW

File: eng/patches/emscripten-version.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+"3.1.7"
```
Comment (radical) [inline eng/patches/emscripten-version.txt]: Can we generate this at build time? And drop the quotes

Comment (radekdoulik) [reply eng/patches/emscripten-version.txt]: This is copy of `emsdk/upstream/emscripten/emscripten-version.txt`. Maybe we can copy it from the docker image during the build.

Comment (radekdoulik) [reply eng/patches/emscripten-version.txt]: I copy it from `$(EMSDK_PATH)` now. The quotes are now stripped with sed on linux build https://github.com/dotnet/icu/pull/197/files#diff-3ef419c17802b9f6582ae68e20648c853b295a742f5437981b8f0b32f2b4aff2R35-R38. Do you know how to strip them on windows or with msbuild?

Comment (radekdoulik) [reply eng/patches/emscripten-version.txt]: I think an inline c# msbuild task might do it here. Will try it.

Comment (radekdoulik) [reply eng/patches/emscripten-version.txt]: Ah, the `CodeTaskFactory` is not available on dotnet msbuild. So used `ReadLinesFromFile`/`WriteLinesToFile` instead.

**Review by lewing — APPROVED** [review]

**Pull request merged**